contract/views/contract.xml

```diff
@@ -0,0 +1,310 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <!--Main FORM view-->
+    <record id="contract_contract_form_view" model="ir.ui.view">
+        <field name="name">contract.contract form view (in contract)</field>
+        <field name="model">contract.contract</field>
+        <field name="arch" type="xml">
+            <form>
+                <header>
+                    <button name="action_contract_send"
+                            type="object"
+                            string="Send by Email"
+                            groups="base.group_user"/>
+                    <button name="recurring_create_invoice"
+                            type="object"
+                            attrs="{'invisible': [('create_invoice_visibility', '=', False)]}"
+                            string="Create invoices"
+                            groups="base.group_no_one"/>
+                </header>
+                <sheet string="Contract">
+                    <div class="oe_button_box" name="button_box">
+                        <button class="oe_stat_button" type="object"
+                                name="toggle_active" icon="fa-archive">
+                            <field name="active" widget="boolean_button"
+                                   options="{&quot;terminology&quot;: &quot;archive&quot;}"/>
+                        </button>
+                        <button name="action_show_invoices"
+                                type="object" icon="fa-list"
+                                class="oe_stat_button">
```

Comment:
i think you should add this context `context="{'default_type': 'out_invoice'}"`.